### PR TITLE
change: ETCM-8949 use permissioned candidates offchain in wizard

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

Uses the Rust implementation of the upsert-permissioned-candidates operation instead of relying on the `pc-contracts-cli` in the wizard.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages.
- [ ] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

**Comment by AmbientTea** [comment]
Upsert-permissioned-cadndidates has a bug which needs to be fixed before this PR can be merged.

File: toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs [hunk 1]
```diff
@@ -90,12 +100,19 @@ fn ariadne_parameters_are_on_main_chain_no_updates() {
...
+			UtxoId::default(),
+			new_d_parameter(),
+			payment_signing_key(),
+			Ok(Some(McTxHash::default())),
+		)
+		.with_upsert_permissioned_candidates(
+			genesis_utxo(),
+			&initial_permissioned_candidates(),
+			payment_signing_key(),
+			Ok(Some(McTxHash::default())),
```
Comment (LGLO) [inline toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs:114]: Perhaps it is logged only, but please use different values in tests. I mean upsert_d_param cannot return the same as upsert_permissioned_candidates

File: toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs [hunk 2]
```diff
@@ -230,40 +254,35 @@ fn prompt_permissioned_candidates_update_io(choice: bool) -> MockIO {
...
-		MockIO::file_read("partner-chains-cli-chain-config.json"),
-		MockIO::run_command_with_result(
-			"./pc-contracts-cli update-permissioned-candidates --remove-all-candidates --network testnet --add-candidate 020a1091341fe5664bfa1782d5e04779689068c916b04cb365ec3153755684d9a1:d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d:88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee --add-candidate 0390084fdbf27d2b79d26a4f13f0ccd982cb755a661969143c37cbc49ef5b91f27:8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48:d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69 --genesis-utxo 0000000000000000000000000000000000000000000000000000000000000000#0 --kupo-host localhost --kupo-port 1442  --ogmios-host localhost --ogmios-port 1337  --payment-signing-key-file payment.skey",
-			Err(anyhow::anyhow!("Something went wrong"))
+		prompt_with_default_and_save_to_existing_file(
+			CARDANO_PAYMENT_SIGNING_KEY_FILE,
+			Some("payment.skey"),
+			"payment.skey",
 		),
+		MockIO::file_read("payment.skey"),
```
Comment (LGLO) [inline toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs:285]: I think few lines below there is now unused `establish_pc_contracts_cli_config_io`, please remove it

File: toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs (hunk 1 above)
Comment (AmbientTea) [reply toolkit/partner-chains-cli/src/setup_main_chain_state/tests.rs]: changed

**Review by LGLO — APPROVED** [review]

**Comment by LGLO** [comment]
I admit that I've added a lot of my changes.
I've tested D-param and permissioned candidates through wizards by creating them with two different genesis utxo and updating them in total over 10 times.

**Pull request merged**